src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -188,30 +321,59 @@ private int NonParamsParameterCount(IMethodSymbol method)
                 ? existingParameters.Count
                 : argumentList.IndexOf(argument);
 
-            AddParameter(
-                syntaxFacts, editor, methodDeclaration, argument,
-                insertionIndex, parameterDeclaration, cancellationToken);
+            if (method.IsExtensionMethod)
+            {
+                insertionIndex++;
+            }
 
-            var newRoot = editor.GetChangedRoot();
-            var newDocument = methodDocument.WithSyntaxRoot(newRoot);
+            return (parameterDeclaration, insertionIndex);
+        }
 
-            return newDocument;
+        private static ITypeParameterSymbol CreateTypeParameterSymbol(IMethodSymbol method)
+        {
+            var exisitingTypeParameter = method.GetAllTypeParameters();
+            var typeParameterName = NameGenerator.GenerateUniqueName("T", n => exisitingTypeParameter.All(symbol => symbol.Name != n));
+            return CodeGenerationSymbolFactory.CreateTypeParameterSymbol(typeParameterName);
         }
 
+        private static ImmutableArray<ITypeParameterSymbol> GetTypeParameterOfClassOrMethod(IMethodSymbol method)
+            => method.IsConstructor()
+                ? (method.ContainingSymbol as ITypeSymbol).GetAllTypeParameters()
```

Comment:
if the method is a constructor, you can just do method.ContainingType.GetAllTypeParameters().

Note: GetAllTypePArameters will get type parameters of containing types as well (if the type is nested).  Is that waht you want?  Needs tests for this sort of thing.